src/particle_header.F90

```diff
@@ -217,6 +226,48 @@ subroutine initialize_from_source(this, src, run_CE, energy_bin_avg)
   end subroutine initialize_from_source
 
 !===============================================================================
+! SET_LAST_COORD copies all coordinate levels from coord to last_coord. This is
+! meant to have information about the last cell in the cell_from filter
+!===============================================================================
+
+  subroutine set_last_coord(this)
+
+    class(Particle), intent(inout) :: this
+    integer :: i
+
+    this % last_n_coord = this % n_coord
+
+    ! copy all information at all coordinate levels
+    do i = 1, this % n_coord
+      call this % last_coord(i) % copy_coord(this % coord(i))
+    end do
```

Comment:
Might just be simpler to just do `this % last_coord = this % coord`. Although it would result in all the xyz and uvw data getting copied too, the compiler might be able to better optimize a straight memory copy.